controller/konnect/ops/ops.go

```diff
@@ -161,8 +165,17 @@ func Create[
 		} else {
 			SetKonnectEntityProgrammedConditionFalse(e, consts.KonnectEntitiesFailedToCreateReason, err.Error())
 		}
+
+	case errors.As(err, &errSDK):
+		switch {
+		case ErrorIsSDKErrorTypeField(err):
+			SetKonnectEntityProgrammedConditionFalse(e, consts.KonnectEntitiesFailedToCreateReason, errSDK.Error())
+		default:
+			SetKonnectEntityProgrammedConditionFalse(e, consts.KonnectEntitiesFailedToCreateReason, err.Error())
+		}
```

Comment:
Why do we need this as a separate case while it's doing the same as the `err != nil` one?

Reply:
That makes sense. I thought initially that we could use this branch here for handling different error types but let's do that when it comes to that. a80100866c5e19ffe540136f34f7f4ee9077521a removes the nested `switch`.